lib/private/legacy/util.php

```diff
@@ -1276,7 +1276,11 @@ public function createHtaccessTestFile(\OCP\IConfig $config) {
 	 * @return bool
 	 */
 	public static function isSetLocaleWorking() {
-		\Patchwork\Utf8\Bootup::initLocale();
+		if (\basename('§') === '') {
+			\setlocale(LC_ALL, 'C.UTF-8', 'C');
+			\setlocale(LC_CTYPE, 'en_US.UTF-8', 'fr_FR.UTF-8', 'es_ES.UTF-8', 'de_DE.UTF-8', 'ru_RU.UTF-8', 'pt_BR.UTF-8', 'it_IT.UTF-8', 'ja_JP.UTF-8', 'zh_CN.UTF-8', '0');
+		}
+		
 		if (\basename('§') === '') {
 			return false;
 		}
```

Comment:
```suggestion
		if (\basename('§') === '') {
			\setlocale(LC_ALL, 'C.UTF-8', 'C');
			\setlocale(LC_CTYPE, 'en_US.UTF-8', 'fr_FR.UTF-8', 'es_ES.UTF-8', 'de_DE.UTF-8', 'ru_RU.UTF-8', 'pt_BR.UTF-8', 'it_IT.UTF-8', 'ja_JP.UTF-8', 'zh_CN.UTF-8', '0');
			return false;
		}
```

Can this all go in a single `if` statement?

Reply:
I guess not, we cannot know if these lang packages are installed on the machine

Reply:
@phil-davis beautified the code but a single if statement is not possible, after calling setLocal this could be evaluate to true,